ui/controls.js

```diff
@@ -788,13 +788,17 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
   /** @export */
   async togglePiP() {
     try {
-      // If you were fullscreen, leave fullscreen first.
-      if (this.isFullScreenEnabled()) {
-        await this.exitFullScreen_();
-      }
       if (this.shouldUseDocumentPictureInPicture_()) {
+        // If you were fullscreen, leave fullscreen first.
```

Comment:
It's not clear to me why copying this to two places inside these branches is better than doing it outside the branches.  It should be the same, right?

In third branch, where we're already in PIP, it shouldn't matter that we run this "exit fullscreen" logic because isFullScreenEnabled_() should be false, right?

Reply:
For the new PiP, you need to exit full screen first and wait for the action to complete, but with the old PiP this isn't necessary and in fact in Safari, waiting for it to complete fails.

Reply:
I'm going to need that in comments here.  Otherwise, I would look at this code a few months from now, and think to myself exactly what I said above: "In third branch, where we're already in PIP, it shouldn't matter that we run this exit fullscreen logic... we should do it outside the branches."